packages/create-react-app/index.js

```diff
@@ -39,6 +39,30 @@
 'use strict';
 
 var chalk = require('chalk');
+var path = require('path');
+var semver = require('semver');
+
+function checkNodeVersion(packageJsonPath) {
+  var packageJson = require(packageJsonPath);
+  if (!packageJson.engines || !packageJson.engines.node) {
+    return;
+  }
+
+  if (!semver.satisfies(process.version, packageJson.engines.node)) {
+    console.error(
+      chalk.red(
+        'You are running Node %s.\n' +
+        'Create React App requires Node %s or higher. \n' +
+        'Please update your version of Node.'
+      ),
+      process.version,
+      packageJson.engines.node
+    );
+    process.exit(1);
+  }
+}
+
+checkNodeVersion(path.resolve(__dirname, 'package.json'));
 
 var currentNodeVersion = process.versions.node
 if (currentNodeVersion.split('.')[0] < 4) {
```

Comment:
This allows us to check the minimal supported version asap for the whole cra to work.
The problem was that it took ~10 minutes with node v0.10 to get to an halt with error, using a lot of resources meanwhile as it still installs all the dependencies, even thou it warns about the problem.